### DISCUSSION TITLE
a post with redirect: link broken?

### DISCUSSION THREAD

**Discussion by ckassung** [alshedivat/al-folio#1818] [Q&A] [answered]
Dear @,
 
if I access the blog entry "a post with redirect" not directly from the blog section via
https://alshedivat.github.io/al-folio/assets/pdf/example_pdf.pdf
but from a tag (2022) the link is
https://alshedivat.github.io/al-folio/blog/2022/redirect/
and does not show the pdf. Would be great if anybody could fix it.

Best, Chris

**Comment by asboyer** [comment]
I agree. While you click on the redirect hyperlink, it just is a link to the new page. When you visit the physical page, it only redirects to the home page.

**Reply by george-gca** [reply]
What do you mean? This is currently working on the latest version. Check clicking in the `a post with redirect` in the [blog index](https://alshedivat.github.io/al-folio/blog/page/4/index.html) and then in the [year tag](https://alshedivat.github.io/al-folio/blog/2022/). They both take you to the pdf.

**Reply by asboyer** [reply]
Yes. Do you know why that happens? Its because in both of those cases, when the title is being displayed on those pages (both the blog index and the year summary page), the "redirect" is just the link for the post. Here is the code that defines this behavior.
```html
  {% if post.redirect == blank %}
    <a class="post-title" href="{{ post.url | relative_url }}">{{ post.title }}</a>
  {% elsif post.redirect contains '://' %}
    <a class="post-title" href="{{ post.redirect }}" target="_blank">{{ post.title }}</a>
    <svg width="2rem" height="2rem" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
      <path d="M17 13.5v6H5v-12h6m3-3h6v6m0-6-9 9" class="icon_svg-stroke" stroke="#999" stroke-width="1.5" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path>
    </svg>
  {% else %}
    <a class="post-title" href="{{ post.redirect | relative_url }}">{{ post.title }}</a>
  {% endif %}
```

**Reply by asboyer** [reply]
The real problem here is when you actually visit the page for the post (https://alshedivat.github.io/al-folio/blog/2022/redirect/) after 3 seconds you will be redirected to the HOME page, rather than the page that it is supposed to be directed to. This is the problem. Where is the code that is responsible for this behavior? Something is telling the page to direct it to the home page, I want the page to direct to the actual specified redirect url.

**Reply by asboyer** [reply, marked as answer]
Solved it, will upload a PR shortly

**Reply by asboyer** [reply]
https://github.com/alshedivat/al-folio/pull/2157
